test/subset-band-name/utilities.py

```diff
@@ -0,0 +1,118 @@
+""""
+Common utility functions used by the subset-band-name regression tests.
+"""
+
+import os
+from pyhdf.SD import SD, SDC
+import numpy
+
+from harmony import Client, Request
+from harmony.harmony import ProcessingFailedException
+
+
+def submit_and_download(
+    harmony_client: Client, request: Request, file_indicator: str
+) -> str:
+    """Submit a Harmony request via a `harmony-py` client. Wait for the
+    Harmony job to finish, then download the results to the specified file
+    path.
+
+    """
+    downloaded_filenames = []
+    output_filename = None
+
+    try:
+        job_id = harmony_client.submit(request)
+
+        print(f"Job ID: {job_id}")
+
+        for filename in [
+            file_future.result()
+            for file_future in harmony_client.download_all(job_id, overwrite=True)
+        ]:
+
+            print(f'Downloaded: {filename}')
+            downloaded_filenames.extend([filename])
+
+            # make a copy to /workdir/output (debugging)
+            # output_filename = f'/workdir/output/{filename.split("/")[-1]}'
+            # os.system(f'cp {filename} {output_filename}')
+
+        for filename in downloaded_filenames:
+            if file_indicator in filename:
+                output_filename = filename
+                print(f'Saved output to: {output_filename}')
+            else:
+                print(f'Could not save output to: {filename}')
+
+    except ProcessingFailedException as exception:
+        print_error('Harmony request failed to complete successfully.')
+        raise exception
+
+    return output_filename
+
+
+def get_dim_sizes(file: str) -> list[int]:
+    ydim = None
+    xdim = None
+    file_SD = SD(file, SDC.READ)
+    datasets = file_SD.datasets()
+    if len(datasets) > 0:
+        dataset_names = datasets.keys()
+        for name in dataset_names:
+            dataset = file_SD.select(name)
+            dims = dataset.dimensions(full=1)
+            ydim = list(dims.values())[0][0]
+            xdim = list(dims.values())[1][0]
+
+    file_SD.end()
+    return [ydim, xdim]
+
+
+def get_sds_data(file: str):
+    sds_data = None
+    file_SD = SD(file, SDC.READ)
+    datasets = file_SD.datasets()
+
+    if len(datasets) > 0:
+        dataset_names = datasets.keys()
+        for name in dataset_names:
+            if name == 'Cloud_Mask':
+                print('Obtaining ', name, ' data')
+                dataset = file_SD.select(name)
+                sds_data = dataset.get()
```

Comment:
I think you could simplify this a fair bit, as you can iterate through keys of a dictionary just with `for key in dictionary:`:

```
for name in file_SD.datasets():
    if name == 'Cloud_Mask':
        print('Obtaining ', name, ' data')
        dataset = file_SD.select(name)
        sds_data = dataset.get()
```

(If the length of `file_SD.datasets()` is 0, then this loop becomes a no-op, so you don't need to check the length of the dictionary).

Reply:
done

Reply:
I'm going to resolve this thread in favour of the continued conversation [here](https://github.com/nasa/harmony-regression-tests/pull/95/files#r1803900933).